### PR TITLE
Add foreign keys to existing tables

### PR DESCRIPTION
**Pull request by pixeltrix**
Rails 4.2 added built-in support for foreign keys and referential integrity so we should take advantage of this to ensure that our database remains consistent even under heavy load.

### CODE REVIEW

File: spec/models/signature_spec.rb [hunk 1]
```diff
@@ -79,84 +79,81 @@
...
+          expect(signature).to have_valid(:email)
+        end
+
+        it "ignores extra whitespace at the end of the name" do
+          signature = FactoryGirl.build(:signature, attributes.merge(name: "Suzy Signer "))
+          expect(signature).not_to have_valid(:email)
+        end
+
+        it "ignores extra whitespace at the beginning of the name" do
+          signature = FactoryGirl.build(:signature, attributes.merge(name: "Suzy Signer "))
```
Comment (pixeltrix) [inline spec/models/signature_spec.rb]: This is meant to be at the beginning - ugh :disappointed:

File: app/models/debate_outcome.rb [hunk 2]
```diff
@@ -2,6 +2,5 @@ class DebateOutcome < ActiveRecord::Base
   belongs_to :petition, touch: true
 
   validates :petition, :debated_on, presence: true
-  validates :petition_id, uniqueness: true
```
Comment (h-lame) [inline app/models/debate_outcome.rb]: There's a uniqueness constraint on ``debate_outcomes.petition_id`` so (race conditions with ``validates_uniqueness_of`` notwithstanding) isn't it still worth having this validation?

**Comment by otlaitil** [comment]
Nice! :+1:

**Comment by h-lame** [comment]
Good spot.  Most of these are my fault too as I did the PRs with that added most of these tables.  Will endeavour to pay more attention on the next models I add.  Don't know how I managed to get these wrong, but ``EmailRequestedReceipt`` and ``EmailSentReceipt`` correct.

**Comment by mybits** [comment]
:+1:

**Pull request merged**